src/components/handlers/IngredientForm.tsx

```
@@ -0,0 +1,179 @@
+const IngredientForm: React.FC<IngredientFormProps> = ({ addIngredient, setIngredients, existingIngredients, loading, error }) => {
+  const [newIngredient, setNewIngredient] = useState<NewIngredientState>({
+    name: '',
+    amount: 0,
+    unit: '',
+    id: 0
+  });
+
+  const getProcessedValue = (name: string, value: string): number | string => {
+    if (name === 'amount') {
+      const parsed = Number(value);
+      if (value !== '' && isNaN(parsed)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)